Content.Shared/Movement/EntitySystems/SlowContactsSystem.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using Content.Shared.Movement.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Physics.Dynamics;
+
+namespace Content.Shared.Movement.EntitySystems;
+
+public class SlowContactsSystem : EntitySystem
+{
+    private readonly Dictionary<EntityUid, int> _statusCapableInContact = new();
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<SlowContactsComponent, StartCollideEvent>(OnEntityEnter);
+        SubscribeLocalEvent<SlowContactsComponent, EndCollideEvent>(OnEntityExit);
+        SubscribeLocalEvent<SlowsOnContactComponent, RefreshMovementSpeedModifiersEvent>(MovementSpeedCheck);
+    }
+
+    private void MovementSpeedCheck(EntityUid uid, SlowsOnContactComponent component, RefreshMovementSpeedModifiersEvent args)
+    {
+
+        if (!_statusCapableInContact.ContainsKey(uid) || _statusCapableInContact[uid] <= 0)
+            return;
+
+        if (!EntityManager.TryGetComponent<PhysicsComponent>(uid, out var physicsComponent))
+            return;
+
+        var walkSpeed = 1.0f;
+        var sprintSpeed = 1.0f;
+
+        foreach (var colliding in _physics.GetCollidingEntities(physicsComponent))
+        {
+            var ent = colliding.OwnerUid;
+            if (!EntityManager.TryGetComponent<SlowContactsComponent>(ent, out var slowContactsComponent))
+                continue;
+
+            walkSpeed = Math.Min(walkSpeed, slowContactsComponent.WalkSpeedModifier);
+            sprintSpeed = Math.Min(sprintSpeed, slowContactsComponent.SprintSpeedModifier);
+        }
+
+        args.ModifySpeed(walkSpeed, sprintSpeed);
```

Comment:
Should check physics contacts instead now as it's significantly faster. The only reason this method even exists is because doors can't and it's legacy.